deploy/stash.sh

```diff
@@ -35,37 +86,26 @@ onessl_found() {
   return 1
 }
 
+# download onessl if it does not exist
 onessl_found || {
   echo "Downloading onessl ..."
-  if [[ "$(uname -m)" == "aarch64" ]]; then
-    curl -fsSL -o onessl https://github.com/kubepack/onessl/releases/download/0.10.0/onessl-linux-arm64
-    chmod +x onessl
-    export ONESSL=./onessl
-  else
-    # ref: https://stackoverflow.com/a/27776822/244009
-    case "$(uname -s)" in
-      Darwin)
-        curl -fsSL -o onessl https://github.com/kubepack/onessl/releases/download/0.10.0/onessl-darwin-amd64
-        chmod +x onessl
-        export ONESSL=./onessl
-        ;;
-
-      Linux)
-        curl -fsSL -o onessl https://github.com/kubepack/onessl/releases/download/0.10.0/onessl-linux-amd64
-        chmod +x onessl
-        export ONESSL=./onessl
-        ;;
-
-      CYGWIN* | MINGW* | MSYS*)
-        curl -fsSL -o onessl.exe https://github.com/kubepack/onessl/releases/download/0.10.0/onessl-windows-amd64.exe
-        chmod +x onessl.exe
-        export ONESSL=./onessl.exe
-        ;;
-      *)
-        echo 'other OS'
-        ;;
-    esac
-  fi
+
+  ARTIFACT="https://github.com/kubepack/onessl/releases/download/0.10.0"
```

Comment:
Why are we changing this?

Reply:
1. If we need to change URL we won't need to change for every OS.
2. Make os detection/download part reusable. If we need to download another binary in future, we would be able to reuse .
3. Support more architecture. In future, we may need to support ARM.